enumerables.rb

```diff
@@ -0,0 +1,286 @@
+        end
+      elsif !cond.is_a?(Numeric) and cond.include?(true || false)
+        my_each do |e|
+          if e == cond
+            classflag = true
+            break
+          end
+        end
+      elsif cond != Class and cond.is_a?(Regexp)
+        my_each do |e|
+          if e.is_a?(String)
+            if e.match?(cond)
+              classflag = true
+              break
+            end
+          end
+        end
+      elsif cond.is_a?(Numeric) || cond.is_a?(String)
+        my_each do |e|
+          if cond == e
+            classflag = true
+            break
+          end
+        end
+      end
+    elsif block_given?
+      my_each do |e|
+        if yield(e)
+          classflag = true
+          break
+        end
+      end
+    elsif cond.nil? and !block_given?
+      classflag = if empty?
+              false
+            else
+              true
+            end
+    else
+      false
+    end
+    classflag
+  end
+
+  # my_none Method
+  def my_none?(cond = nil)
+    classflag = true
+    if !cond.nil?
+      if cond.is_a?(Class)
+        my_each do |e|
+          if e.is_a?(cond)
+            classflag = false
+            break
+          end
+        end
+      elsif cond == Numeric
+        my_each do |e|
+          if e.is_a?(Numeric)
+            classflag = false
+            break
+          end
+        end
+      elsif cond.is_a?(Boolean)
+        my_each do |e|
+          if e == cond
+            classflag = false
+            break
+          end
+        end
+      elsif cond != Class and cond.is_a?(Regexp)
+        my_each do |e|
+          if e.is_a?(String)
+            if e.match?(cond)
+              classflag = false
+              break
+            end
+          end
+        end
+      elsif cond.is_a?(Numeric) || cond.is_a?(String)
+        my_each do |e|
+          if cond == e
+            classflag = false
+            break
+          end
+        end
+      end
+
+    elsif block_given?
+      my_each do |e|
+        if yield(e)
+          classflag = false
+          break
+        end
+      end
+    elsif cond.nil? and !block_given?
+      classflag = if to_a.empty?
+        true
+      else
+        false
+      end
+    else
+      true
+    end
+    classflag
+  end
+
+  # my_none Method (another approach), didn't work because we couldn't use yield on a yield...
+  # def my_none_2?(cond = nil)
+  #   res = false
+  #   if !cond.nil?
+  #     inv = self.my_any?(cond)
+  #   elsif block_given?
+  #     block = yield
+  #     inv = self.my_any? {yield.to_s}
+  #   end
+  #   inv == true ? res = false : res = true
+  #   res
+  #   block
+  # end
+
+  # my_count Method
+  def my_count(cond = nil)
+    counter = 0
+    if block_given?
+      my_each do |f|
+        counter += if yield f
+        end
+        counter
+      end
+    elsif cond.nil?
+      length
+    elsif !cond.nil?
+      my_each do |e|
+        counter += 1 if cond == e
+      end
+      counter
+    end
+  end
```

Comment:
- [ ] should have working `my_count` **(covered by automated tests)** identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))

#### my_count

- [ ] my_count returns the number of items in enum through enumeration
```ruby
(1..5).my_count == (1..5).count => 5
```
- my_count is giving an NameError
![image](https://user-images.githubusercontent.com/52209067/95549720-03afeb00-09c5-11eb-8cac-1863fc275f9d.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/95549744-11fe0700-09c5-11eb-8ec1-0a796cccb3b2.png)

- [ ] my_count counts the number of elements yielding a true value if a block is given
```ruby
[5, 4, 1].my_count { |x| x > 2 } == [5, 4, 1].count { |x| x > 2 } => 2
```
- my_count is giving an TypeError
![image](https://user-images.githubusercontent.com/52209067/95549952-60aba100-09c5-11eb-83c3-aca8d535eab4.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/95550000-702aea00-09c5-11eb-9b4f-0b0d33c2c1e3.png)